autocompletecontext.go

```diff
@@ -246,19 +252,25 @@ func (c *auto_complete_context) get_candidates_from_decl(cc cursor_context, clas
 				continue
 			}
 		}
-		b.append_decl(cc.partial, decl.name, decl, class)
+		pkgname := ""
+		if pkg, ok := c.pcache[decl.scope.pkgname]; ok {
+			pkgname = pkg.import_name
+		}
+		b.append_decl(cc.partial, decl.name, pkgname, decl, class)
 	}
 	// propose all children of an underlying struct/interface type
 	adecl := advance_to_struct_or_interface(cc.decl)
 	if adecl != nil && adecl != cc.decl {
 		for _, decl := range adecl.children {
 			if decl.class == decl_var {
-				b.append_decl(cc.partial, decl.name, decl, class)
+				pkgname := c.pcache[decl.name].import_name
+				b.append_decl(cc.partial, decl.name, pkgname, decl, class)
 			}
 		}
 	}
 	// propose all children of its embedded types
-	b.append_embedded(cc.partial, cc.decl, class)
+	pkgname := c.pcache[cc.decl.name].import_name
```

Comment:
gocode itself in various places.
e.g. in here if you try `decl.`
https://github.com/nsf/gocode/blob/f1eef9a6ba005abb145d7b58fdd225e83a3c6a05/autocompletecontext.go#L68-L86 
So I guess it's happening whenever you try to access variables or functions within the same package.